middleware/src/middleware.go

```diff
@@ -159,9 +173,82 @@ func (middleware *Middleware) rpcLoop() {
 	}
 }
 
+// updateCheckLoop repeatedly checks for information about new Base firmware updates
+// When an update is available it's
+func (middleware *Middleware) updateCheckLoop() {
+	// This time is chosen arbitrary, but the time should be not too high for users to be notified
+	// not to long after the update release and not too low to avoid to frequent update checks.
+	const timeBetweenUpdateChecks time.Duration = 30 * time.Minute
```

Comment:
I personally think a 30 minute update check interval is a good compromise between too frequent and to infrequent. But this is chosen arbitrarily. Looking for other suggestions.

Reply:
I think twice a day would suffice as well, if we could trigger a manual check from the App (e.g. if we announce an update), but that's not really important. I'm fine with just checking every 30 minutes.

Reply:
Will leave it at 30 minutes.